compress.go

```diff
@@ -0,0 +1,144 @@
+package iavl
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+type NodeExporter interface {
+	Next() (*ExportNode, error)
+}
+
+type NodeImporter interface {
+	Add(*ExportNode) error
+}
+
+// CompressExporter wraps the normal exporter to apply some compressions on `ExportNode`:
+// - branch keys are skipped
+// - leaf keys are encoded with delta compared with the previous leaf
+// - branch node's version are encoded with delta compared with the max version in it's children
+type CompressExporter struct {
+	inner        NodeExporter
+	lastKey      []byte
+	versionStack []int64
+}
+
+var _ NodeExporter = (*CompressExporter)(nil)
+
+func NewCompressExporter(exporter NodeExporter) NodeExporter {
+	return &CompressExporter{inner: exporter}
+}
+
+func (e *CompressExporter) Next() (*ExportNode, error) {
+	n, err := e.inner.Next()
+	if err != nil {
+		return nil, err
+	}
+
+	if n.Height == 0 {
+		// apply delta encoding to leaf keys
+		n.Key, e.lastKey = deltaEncode(n.Key, e.lastKey), n.Key
+
+		e.versionStack = append(e.versionStack, n.Version)
+	} else {
+		// branch keys can be derived on the fly when import, safe to skip
+		n.Key = nil
+
+		// delta encode the version
+		maxVersion := maxInt64(e.versionStack[len(e.versionStack)-1], e.versionStack[len(e.versionStack)-2])
```

Comment:
just curious about the efficiency of version delta encode?

Reply:
https://github.com/cosmos/iavl/pull/703#issuecomment-1472958077